sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/MessageFluxRequestDrivenCreditFlowIsolatedTest.java

```diff
@@ -0,0 +1,303 @@
+
+import com.azure.core.amqp.AmqpEndpointState;
+import com.azure.core.amqp.AmqpRetryOptions;
+import com.azure.core.amqp.AmqpRetryPolicy;
+import com.azure.core.amqp.FixedAmqpRetryPolicy;
+import org.apache.qpid.proton.message.Message;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.api.parallel.Isolated;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.publisher.TestPublisher;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Execution(ExecutionMode.SAME_THREAD)
+@Isolated
+public class MessageFluxRequestDrivenCreditFlowIsolatedTest {
+    private static final int MAX_RETRY = 3;
+    private static final Duration RETRY_DELAY = Duration.ofSeconds(3);
+    private final AmqpRetryOptions retryOptions = new AmqpRetryOptions().setMaxRetries(MAX_RETRY).setDelay(RETRY_DELAY);
+    private final AmqpRetryPolicy retryPolicy = new FixedAmqpRetryPolicy(retryOptions);
+    private AutoCloseable mocksCloseable;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        mocksCloseable = MockitoAnnotations.openMocks(this);
+    }
+
+    @AfterEach
+    public void teardown() throws Exception {
+        Mockito.framework().clearInlineMock(this);
+
+        if (mocksCloseable != null) {
+            mocksCloseable.close();
+        }
+    }
+
+    @Test
+    @Execution(ExecutionMode.SAME_THREAD)
+    public void initialFlowShouldBeSumOfRequestAndPrefetch() {
+        final int prefetch = 100;
+        final TestPublisher<ReactorReceiver> upstream = TestPublisher.create();
+        final MessageFlux messageFlux = new MessageFlux(upstream.flux(), prefetch, CreditFlowMode.RequestDriven, retryPolicy);
+
+        final ReactorReceiver receiver = mock(ReactorReceiver.class);
+        when(receiver.receive()).thenReturn(Flux.never());
+        when(receiver.getEndpointStates()).thenReturn(activeThenNeverTerminate());
+        when(receiver.closeAsync()).thenReturn(Mono.empty());
+
+        final AtomicLong initialFlow = new AtomicLong();
+        doAnswer(invocation -> {
+            final Object[] args = invocation.getArguments();
+            @SuppressWarnings("unchecked")
+            final Supplier<Long> creditSupplier = (Supplier<Long>) args[0];
```

Comment:
Instead of this casting, we can just use:
`Supplier<Long> creditSupplier = invocation.getArgument(0);`

Reply:
Same with the other ones below.

Reply:
thank you, this helps removing the cast :thumbsup: